src/main/java/org/jpeek/graph/XmlGraph.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.graph;
+
+import com.jcabi.xml.XML;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.cactoos.text.Joined;
+import org.jpeek.skeleton.Skeleton;
+
+/**
+ * Graph implementation built on skeleton.
+ * @since 0.30.9
+ */
+public final class XmlGraph implements Graph {
+
+    /**
+     * List of the nodes of this graph.
+     */
+    private final List<Node> nds;
+
+    /**
+     * Ctor.
+     * @param skeleton XMl representation on whiwh to build the graph
+     * @throws IOException If fails
+     */
+    public XmlGraph(final Skeleton skeleton) throws IOException {
+        this.nds = XmlGraph.build(skeleton);
```

Comment:
@HDouss I don't think it is a good idea to execute code inside constructor.

Reply:
@fanifieiev generally we can execute code if it is to initialize a final field. I think it would be better than to make the field non final and execute the code later. What do you think?

Reply:
@HDouss You can use org.cactoos.Scalar for that case. 
In Cactoos project we have a lot of such cases.
The main idea is to make a late execution.

Reply:
@fanifieiev Thanks.